apps/aecore/src/aec_pinning_agent.erl

```diff
@@ -0,0 +1,129 @@
+%%%-----------------------------------------------------------------------------
+%%% @doc
+%%% Default pinning agent for hyperchains.
+%%% Supported by the Aeternity Foundation
+%%% @author mans.af.klercker@happihacking.se
+%%% @end
+%%%-----------------------------------------------------------------------------
+
+-module(aec_pinning_agent).
+-author("mans.af.klercker@happihacking.se").
+
+%%%=============================================================================
+%%% Export and Defs
+%%%=============================================================================
+
+%% External API
+-export([
+    spawn_for_epoch/3,
+    start/3
+    ]).
+
+-define(SERVER, ?MODULE).
+
+%%%=============================================================================
+%%% API
+%%%=============================================================================
+
+
+spawn_for_epoch(EpochInfo, Contract, LastLeader) ->
+    lager:debug("AGENT: Trying to spawn pinning agent...", []),
+    try
+    case whereis(my_unique_process) of
```

Comment:
Is `?MODULE` a better name for the registered process?

Reply:
Agreed that eventually, ?MODULE is better. Will change to that.

Reply:
Ah, true. I guess that's the standard way of doing these things.